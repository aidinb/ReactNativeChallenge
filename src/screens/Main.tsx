import React from 'react';
import { View, Text, Image, StyleSheet, Dimensions } from 'react-native';
import { observer } from 'mobx-react';
import { COLORS } from '../styles';
import Loading from '../components/Loading';
import ErrorMessage from '../components/ErrorMessage';
import { useStores } from '../stores';

const logo = require('../assets/reactnativelogo.png');

const { width, height } = Dimensions.get('window');

const Main: React.FC = () => {
  const { mainStore } = useStores(); // Access the store

  return (
      <View style={styles.container}>
        <Text style={styles.welcomeText}>Welcome</Text>
        <Image source={logo} style={styles.logo} testID={'logo-image'}/>
        {mainStore.error && <ErrorMessage message={mainStore.error} />}
        {mainStore.loading && <Loading />}
      </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.white,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  welcomeText: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#333',
    marginBottom: 20,
    textAlign: 'center',
    position: 'absolute',
    top: 80,
  },
  logo: {
    width: width * 0.5,
    height: height * 0.2,
    resizeMode: 'contain',
    marginTop: 40,
  },
});

export default observer(Main);
