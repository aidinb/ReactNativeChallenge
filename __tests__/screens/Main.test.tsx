import React from 'react';
import { render } from '@testing-library/react-native';
import Main from '../../src/screens/Main'; // Adjust path if needed

describe('Main Screen', () => {
    it('renders correctly', () => {
        const { toJSON } = render(<Main />);
        expect(toJSON()).toMatchSnapshot();
    });

    it('contains a welcome text', () => {
        const { getByText } = render(<Main />);
        const welcomeText = getByText('Welcome');
        expect(welcomeText).toBeTruthy();
    });

    it('contains an image with the correct source', () => {
        const { getByTestId } = render(<Main />);
        const image = getByTestId('logo-image');
        expect(image.props.source).toEqual(require('../../src/assets/reactnativelogo.png'));
    });
});
