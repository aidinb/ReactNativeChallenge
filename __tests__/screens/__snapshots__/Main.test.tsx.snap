// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Screen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginBottom": 50,
        "marginTop": 50,
        "textAlign": "center",
      }
    }
  >
    Welcome
  </Text>
  <Image
    source={
      {
        "testUri": "../../../src/assets/reactnativelogo.png",
      }
    }
    style={
      {
        "height": 300,
        "resizeMode": "contain",
        "width": 650,
      }
    }
    testID="logo-image"
  />
</View>
`;
